packages/core/src/fhirpath/utils.ts

```diff
@@ -101,20 +101,29 @@ function getTypedPropertyValueWithSchema(
   let resultValue: any = undefined;
   let resultType = 'undefined';
 
-  if (types.length === 1) {
-    resultValue = input.value[path];
+  if (path === 'value[x]') {
+    for (const type of types) {
+      const path2 = 'value' + capitalize(type.code);
+      if (path2 in value) {
+        resultValue = value[path2];
```

Comment:
Actually, I think this block should be compatible with the `else` block below?

Reply:
It would, except the length of `types` for `Extension.value[x]` can have a length of 1, so the `else if (types.length === 1)` would catch it.

Here's an example from [this extension](https://hl7.org/fhir/us/core/STU6.1/StructureDefinition-us-core-birthsex.html)

```javascript
    {
      "id" : "Extension.value[x]",
      "path" : "Extension.value[x]",
      "type" : [{
        "code" : "code"
      }],
      "binding" : {
        "strength" : "required",
        "description" : "Code for sex assigned at birth",
        "valueSet" : "http://hl7.org/fhir/us/core/ValueSet/birthsex"
      },
    // ...Other properties excluded for brevity
    }
```

If we don't want to have the similar contents of the `if` and `else` blocks, it seems like the following would be equivalent to what we have now, but feels weird with that dangling else block...
```javascript
  if (path.endsWith('[x]')) {
    for (const type of types) {
      const path2 = path.replace('[x]', '') + capitalize(type.code);
      if (path2 in value) {
        resultValue = value[path2];
        resultType = type.code;
        break;
      }
    }
  } else if (types.length === 1) {
    resultValue = value[path];
    resultType = types[0].code;
  } else {
    // what to do?
  }
```

Reply:
1. Offline discussions helped me understand the case where profile restricts the `types` array, so that makes sense
2. Note this other PR which may cause a merge conflict here :'( https://github.com/medplum/medplum/pull/3538